Source/CBL_Router.m

```diff
@@ -204,6 +204,17 @@ - (CBLQueryOptions*) getQueryOptions {
     options->group = [self boolQuery: @"group"];
     options->content = [self contentOptions];
 
+    // Stale options (ok or update_after):
+    NSString *stale = [self query: @"stale"];
+    if (stale) {
+        if ([stale isEqualToString:@"ok"])
+            options->indexUpdateMode = kCBLUpdateIndexNever;
+        else if ([stale isEqualToString:@"update_after"])
+            options->indexUpdateMode = kCBLUpdateIndexAfter;
+        else
+            return nil;
```

Comment:
I checked with CouchDB v1.6.0 and it also reported an error when I gave an invalid value.

Returning nil will allow the router handler to report an error which is kCBLStatusBadRequest when the stale option is invalid. The error status is inline with the current implementation when view query params such as startKey, endKey or, bbox is invalid. 

Let me know if you confirm that we shouldn't return the error and should use the default update_before mode for this case.